File: app/src/main/java.backup/com/lumiyaviewer/lumiya/slproto/messages/StartPingCheck.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.google.common.primitives.UnsignedBytes;
import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;

public class StartPingCheck extends SLMessage {
    public PingID PingID_Field = new PingID();

    public static class PingID {
        public int OldestUnacked;
        public int PingID;
    }

    public StartPingCheck() {
        this.zeroCoded = false;
    }

    public int CalcPayloadSize() {
        return 6;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleStartPingCheck(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.put((byte) 1);
        packByte(byteBuffer, (byte) this.PingID_Field.PingID);
        packInt(byteBuffer, this.PingID_Field.OldestUnacked);
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.PingID_Field.PingID = unpackByte(byteBuffer) & UnsignedBytes.MAX_VALUE;
        this.PingID_Field.OldestUnacked = unpackInt(byteBuffer);
    }
}

File: Linkpoint/src/main/java/com/linkpoint/slproto/messages/StartPingCheck.kt
package com.linkpoint.slproto.messages

import com.google.common.primitives.UnsignedBytes
import com.linkpoint.slproto.SLMessage
import java.nio.ByteBuffer

class StartPingCheck : SLMessage() {
    val PingID_Field = PingID()

    class PingID {
        var PingID: Int = 0
        var OldestUnacked: Int = 0
    }

    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int = 6

    override fun Handle(handler: SLMessageHandler) {
        handler.HandleStartPingCheck(this)
    }

    override fun PackPayload(buffer: ByteBuffer) {
        buffer.put(1.toByte())
        packByte(buffer, PingID_Field.PingID.toByte())
        packInt(buffer, PingID_Field.OldestUnacked)
    }

    override fun UnpackPayload(buffer: ByteBuffer) {
        PingID_Field.PingID = unpackByte(buffer).toInt() and UnsignedBytes.MAX_VALUE.toInt()
        PingID_Field.OldestUnacked = unpackInt(buffer)
    }
}